### PR TITLE
Update "GDScript reference" (operators and `match`)

### PR DESCRIPTION
**Pull request by dalexeev**
Closes godotengine/godot#74462.

### CODE REVIEW

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 1]
```diff
@@ -1190,6 +1210,9 @@ Basic syntax::
         [pattern](s):
             [block]
 
+.. warning::
+
+    ``match`` is more type strict than the ``==`` operator. For example ``1`` will NOT match ``1.0``. The only exception is ``String`` vs ``StringName`` matching.
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_basics.rst:1215]:
```suggestion
    ``match`` is more type-strict than the ``==`` operator. For example, ``1`` will **not** match ``1.0``. The only exception is ``String`` vs ``StringName`` matching.
```

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 2]
```diff
@@ -1198,6 +1221,10 @@ Basic syntax::
 3. Remove any ``break``\ s.
 4. Change ``default`` to a single underscore.
 
+.. note::
+
+    The special ``continue`` behavior in ``match`` supported in 3.x was removed prior to the 4.0 release.
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_basics.rst:1226]: Make the wording more explicit:

```suggestion
    The special ``continue`` behavior in ``match`` supported in 3.x was removed in Godot 4.0.
```

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 3]
```diff
@@ -503,7 +523,7 @@ There are two ways to represent an escaped Unicode character above 0xFFFF:
 Also, using ``\`` followed by a newline inside a string will allow you to continue it in the next line, without
 inserting a newline character in the string itself.
 
-GDScript also supports :ref:`doc_gdscript_printf`.
+GDScript also supports :ref:`Format Strings <doc_gdscript_printf>`.
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_basics.rst:526]: "Format strings" isn't a proper word per se, so I wouldn't capitalize it:

```suggestion
GDScript also supports :ref:`format strings <doc_gdscript_printf>`.
```

**Review by mhilbrunner — APPROVED** [review]

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 4]
```diff
@@ -1824,8 +1875,8 @@ Our ``BattleLog`` node receives each element in the binds array as an extra argu
         label.text += character_name + " took " + str(damage) + " damage."
 
 
-Awaiting for signals
-~~~~~~~~~~~~~~~~~~~~
+Awaiting for signals or coroutines
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The ``await`` keyword can be used to create `coroutines <https://en.wikipedia.org/wiki/Coroutine>`_
 which waits until a signal is emitted before continuing execution. Using the ``await`` keyword with a signal or a
```
Comment (mhilbrunner) [inline tutorials/scripting/gdscript/gdscript_basics.rst:1882]:
```suggestion
which wait until a signal is emitted before continuing execution. Using the ``await`` keyword with a signal or a
```

**Comment by mhilbrunner** [comment]
Thank you! Merged.

**Comment by mhilbrunner** [comment]
Cherry-picked to `4.0`/`stable` docs branch.

**Pull request merged**